tools/CorrelationTestbed/Test-CorrelationInSandbox.ps1

```diff
@@ -0,0 +1,342 @@
+"@
+    }
+
+
+    $bootstrapPs1FileName = 'Bootstrap.ps1'
+    $bootstrapPs1Content | Out-File (Join-Path $tempFolder $bootstrapPs1FileName) -Force
+
+    # Create Wsb file
+
+    $bootstrapPs1InSandbox = Join-Path -Path $desktopInSandbox -ChildPath (Join-Path -Path $tempFolderName -ChildPath $bootstrapPs1FileName)
+    $tempFolderInSandbox = Join-Path -Path $desktopInSandbox -ChildPath $tempFolderName
+    $exePathInSandbox = Join-Path -Path $desktopInSandbox -ChildPath "InstallAndCheckCorrelation"
+
+    $devPackageInSandbox = Join-Path -Path $desktopInSandbox -ChildPath "DevPackage"
+    $devPackageXMLFragment = ""
+
+    if ($UseDev)
+    {
+      $devPackageXMLFragment = @"
+      <MappedFolder>
+          <HostFolder>$DevPackagePath</HostFolder>
+          <SandboxFolder>$devPackageInSandbox</SandboxFolder>
+      </MappedFolder>
+"@
+    }
+
+    $regFileDirInSandbox = Join-Path -Path $desktopInSandbox -ChildPath "RegFiles"
+    $regFileDirXMLFragment = ""
+
+    if ($RegFileDirectory)
+    {
+      $regFileDirXMLFragment = @"
+      <MappedFolder>
+          <HostFolder>$RegFileDirectory</HostFolder>
+          <SandboxFolder>$regFileDirInSandbox</SandboxFolder>
+          <ReadOnly>true</ReadOnly>
+      </MappedFolder>
+"@
+    }
+
+    $sandboxTestWsbContent = @"
+<Configuration>
+    <MappedFolders>
+    <MappedFolder>
+        <HostFolder>$tempFolder</HostFolder>
+        <ReadOnly>true</ReadOnly>
+    </MappedFolder>
+    <MappedFolder>
+        <HostFolder>$ExePath</HostFolder>
+        <SandboxFolder>$exePathInSandbox</SandboxFolder>
+        <ReadOnly>true</ReadOnly>
+    </MappedFolder>
+    $devPackageXMLFragment
+    $regFileDirXMLFragment
+    <MappedFolder>
+        <HostFolder>$outPath</HostFolder>
+    </MappedFolder>
+    </MappedFolders>
+    <LogonCommand>
+    <Command>PowerShell Start-Process PowerShell -WindowStyle Maximized -WorkingDirectory '$tempFolderInSandbox' -ArgumentList '-ExecutionPolicy Bypass -NoExit -NoLogo -File $bootstrapPs1InSandbox'</Command>
+    </LogonCommand>
+</Configuration>
+"@
+
+    $sandboxTestWsbFileName = 'SandboxTest.wsb'
+    $sandboxTestWsbFile = Join-Path -Path $tempFolder -ChildPath $sandboxTestWsbFileName
+    $sandboxTestWsbContent | Out-File $sandboxTestWsbFile -Force
+
+    Write-Host @"
+--> Starting Windows Sandbox, and:
+    - Mounting the following directories:
+        - $tempFolder as read-only
+        - $outPath as read-write
+    - Installing WinGet
+    - Installing the package: $packageIdentifier
+    - Comparing ARP Entries
+"@
+
+    Write-Host
+
+    WindowsSandbox $SandboxTestWsbFile
+
+    $outputFileBlockerPath = Join-Path $outPath "done.txt"
+
+    while (-not (Test-Path $outputFileBlockerPath))

```

Comment:
I would add a timeout here. When I tried doing something similar, I saw many cases where the app would still require interaction and it would block the script until I went to click something.

I think that shouldn't happen, but it seems the msstore source doesn't always gives us the right arguments for the installers :(